op-program/client/l2/oracle_test.go

```diff
@@ -0,0 +1,124 @@
+package l2
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/testutils"
+	"github.com/ethereum-optimism/optimism/op-program/client/mpt"
+	"github.com/ethereum-optimism/optimism/op-program/preimage"
+)
+
+func mockPreimageOracle(t *testing.T) (po *PreimageOracle, hintsMock *mock.Mock, preimages map[common.Hash][]byte) {
+	// Prepare the pre-images
+	preimages = make(map[common.Hash][]byte)
+
+	hintsMock = new(mock.Mock)
```

Comment:
IMO the number of hints called for isn't part of the correctness conditions & this would be a really easy function to fake rather than mock.

Reply:
I think it is important in this case to be specific about what hints are requested (ideally we'd also assert that the hint must precede requesting the preimages it needs since the host fetching code will depend on that).

Reply:
Wanted to be very explicit about which hints were being expected, it's not really complicated functionality, just need to make sure something is called so we know the pre-images can be prepared as necessary.